common/brave_channel_info_linux.cc

```diff
@@ -16,11 +17,18 @@ std::string GetChannelSuffixForDataDir() {
 
   // Chrome doesn't support canary channel on linux.
   if (modifier == "unstable")  // linux version of "dev"
+    modifier = "dev";
```

Comment:
if we use defines in a gni config as suggested https://github.com/brave/brave-core/pull/208/files#r199649672 we can remove some or all of this logic and just return the value from the gn define

Reply:
otherwise I'm concerned about the two getting out of sync

Reply:
There are many unit test and browser test that uses this modifier.
So, if someone make inconsistency between c++ and gn, it can be easily detected. WDYT?

Reply:
Unified all linux channel infos to brave/build/linux and used it in c++.
PTAL.